### ISSUE TITLE
jCrop error

### ISSUE THREAD

**Issue by pitangaweb** [Jako/ImagePlus#60]
On MODX 2.5.7 with Image+ 2.6.3-pl, I get the following error when Redactor 2.3.0-pl is installed:

`TypeError: this.$image.Jcrop is not a function. (In 'this.$image.Jcrop(a,function(a){return function(){a.cropper=this,this.setOptions({outerImage:a.window.getOuterImageUrl(),bgOpacity:.5})}}(this))', 'this.$image.Jcrop' is undefined) -  imageplus.min.js:11:5344`

Obviously the cropping function does not work.

It disappears when I uninstall Redactor.

**Comment by Mark-H** [comment]
Perhaps a jquery conflict?

**Comment by Jako** [comment]
Plugin order could maybe help. Should be a jquery overwrite issue.

**Comment by pitangaweb** [comment]
Is there something I could try as a quick fix ?

**Comment by Mark-H** [comment]
Redactor grabs its own version of jQuery (`$red`) with noconflict mode, which usually helps prevent stuff like this. Does Image+ attempt to get noconflict as well?

**Comment by Jako** [comment]
Yes, since version 2.3.3. 

See: https://github.com/Jako/ImagePlus/blob/6cba12c356799a07f1d01077197bb364c1527782/core/components/imageplus/model/imageplus/imageplus.class.php#L196-L199
https://github.com/Jako/ImagePlus/blob/527ee2fa0eadcf0e4af71da9d6bd3c085a16a4c1/source/js/mgr/imageplus.jquery.imagecrop.js#L51

**Comment by Jako** [comment]
Is Redactor using JCrop too?

**Comment by bvandorp** [comment]
I can confirm this behavior, is this something that Image+ has to fix or Redactor? 
Don't like to choose between you two :P

**Comment by woutercox** [comment]
The same error still exists when you use a custom 'Redactor TV' together with ImagePlus in the same document. Without the redactor TV set, cropping works.

**Comment by brain2002** [comment]
@woutercox +1

**Comment by Jako** [comment]
I hope I get this sorted in december. I am quite busy at the moment.

**Comment by Jako** [comment]
@Mark-H How do you insert the scripts into the page? modx->regClient methods or modx->controller->add methods?

**Comment by Mark-H** [comment]
Redactor uses `$modx->event->output($html)` from `OnRichTextEditorInit`. The `$html` is sourced from a template that loads the necessary script(s) and immediate runs the noConflict code. 

````
<script src="[[+assetsUrl]]redactor-2.2.3.min.js"></script>
[[+plugin_files]]
[[+langFile:notempty=`<script type="text/javascript" src="[[+langFile]]"></script>`]]
<script type="text/javascript">
    var $red = (typeof($red) != 'undefined') ? $red : $.noConflict();
    // a lot more code
</script>
````

The redactor-2.2.3.min.js file contains all dependencies including a (rather old, I have to admit) version of jQuery. No jCrop. 

Feel free to grab a [free dev license](https://www.modmore.com/free-development-licenses/) if you want to try loading them side by side..

**Comment by Jako** [comment]
Should be fixed with 2.8.0

It occured since the I the jQuery.noConflict() variable was initialized too late.

**Comment by Jako** [comment]
You could you try the 2.8.0-rc1: https://github.com/Jako/ImagePlus/raw/master/_packages/imageplus-2.8.0-rc1.transport.zip
